workflow/metrics/scripts/metrics/morans_i.py

```diff
@@ -144,3 +144,80 @@ def morans_i_genes(adata, output_type, gene_set, **kwargs):
         print(name, score, flush=True)
     
     return scores, metric_names
+
+
+def morans_i_genescore(adata, output_type, gene_set, **kwargs):
+    adata = dask_compute(adata, layers='X')
+    
+    metric = "M's I gene score"
+    metric_names = []
+    scores = []
+    
+    for set_name, gene_list in tqdm(gene_set.items(), desc='Compute Moran\'s I for gene sets'):
+        gene_list = parse_gene_names(adata, gene_list)
+        
+        gene_score_name = f'gene_score:{set_name}'
+        # random_gene_score_name = f'random_gene_scores:{len(gene_list)}'
+        
+        if gene_score_name not in adata.obs.keys():
+            raise ValueError(f'Gene score {gene_score_name} not found in adata.obs')
+        
+    
+        score = _morans_i(adata, covariate=gene_score_name)
+        
+        # random_gene_score = np.nanmean(
+        #     _morans_i(adata, covariate=adata.obsm[random_gene_score_name])
+        # )
+        
+        scores.extend([
+            score,
+            # score - random_gene_score,
+            # random_gene_score
+        ])
+        
+        metric_names.extend([
+            f'{metric}:{set_name}',
+        ])
+        
+    scores = [max(s, 0) for s in scores]  # ensure score is positive
+    for name, score in zip(metric_names, scores):
+        print(name, score, flush=True)
+    
+    return scores, metric_names
+
+
+def morans_i_genescore_random(adata, output_type, gene_set, **kwargs):
```

Comment:
The random score was initially thought to be for normalising the gene-score based Moran's I, but not as a standalone metric per gene set. Could you make normalising by random score an optional parameter for the `morans_i_genescore` function? This doesn't need to be configurable by the user yet.